js/analyzer.js

```diff
@@ -239,42 +217,11 @@ YoastSEO.Analyzer.prototype.keywordCount = function() {
  * @returns resultObject
  */
 YoastSEO.Analyzer.prototype.subHeadings = function() {
-	var result = [ { test: "subHeadings", result: { count: 0, matches: 0 } } ];
+	var getSubheadings = require ( "./analyses/getSubheadings.js" );
 
-	//matches everything from H1-H6 openingtags untill the closingtags.
-	var matches = this.preProcessor.__store.cleanTextSomeTags.match( /<h([1-6])(?:[^>]+)?>(.*?)<\/h\1>/ig );
-	if ( matches !== null ) {
-		result[ 0 ].result.count = matches.length;
-		result[ 0 ].result.matches = this.subHeadingsCheck( matches );
-	}
-	return result;
-};
+	var result = [ { test: "subHeadings", result: getSubheadings( this.preProcessor.__store.originalText, this.config.keyword ) } ];
```

Comment:
I thought the preprocessor could be removed after these changes, we should at least be able to remove it completely from the analyser.